src/pending_transactions.erl

```diff
@@ -117,12 +117,50 @@ add_transaction(Transaction, PendingTx)
     end.
     
 
-%% @doc Get list of all the players in the container.
+%% doc Get list of all the players in the container.
 
-get_players(_) ->
-    ok.
+%% get_all_players(#pending_tx{player_map = PlMap}) ->
+%%     maps:keys(PlMap).
 
-%% @doc Get all the pending transactions after a point.
-
-get_pending_transactions(_, _) ->
-    ok.
+%% @doc Returns pending transactions in the correct order.
+%% 
+%% For each player P, transactions will start at PlayersStartingNonceMap[P]
+%% and will have consecutive nonces (no gaps).
+
+get_pending_transactions(PlayersStartingNonceMap, PendingTx) 
+  when is_map(PlayersStartingNonceMap),
+       is_record(PendingTx, pending_tx) 
+       ->
+
+    PlMap = PendingTx#pending_tx.player_map,
+
+    %% (Key, Value, Acc) applied to PlMap
+    FoldFn = fun (PlayerId, TxMap, TransactionListAcc) ->
+		     R = maps:find(PlayerId, PlayersStartingNonceMap),
+		     case R of
+			 {ok, StartingNonce} -> ok;
+			 error -> StartingNonce = 0
+		     end,
+		     collect_transactions_from_player(TxMap, StartingNonce, TransactionListAcc)
+	     end,
+
+    % unsorted list of tuples {Counter, Transaction}
+    TransactionList = maps:fold(FoldFn, [], PlMap),
+
+
+    TupleCompareFn = fun({Counter1, _}, {Counter2, _}) -> Counter1 =< Counter2 end,
+    TupleMapFn = fun({_, Transaction}) -> Transaction end,
+
+    lists:map(TupleMapFn, lists:sort(TupleCompareFn, TransactionList)).
+
+%% extracts transactions from a player, in sequential order, starting at Nonce
+collect_transactions_from_player(TxMap, Nonce, TransactionListAcc) ->
```

Comment:
:o

Reply:
I would have some suggestions if this were haskell XD

Reply:
that's actually pretty neat

Reply:
I came up with a better one 🙀 

unfoldr (\acc -> map (,acc+1) (lookup acc txmap)) nonce

(using TupleSections extension)

Reply:
unfold 😱

Reply:
incoming potato haskell pull request

Reply:
but it may be a while because I'm trying to implement unfoldr in erlang without recursion 😱